internal/transport/flowcontrol.go

```diff
@@ -66,13 +70,15 @@ func (w *writeQuota) get(sz int32) error {
 }
 
 func (w *writeQuota) realReplenish(n int) {
-	sz := int32(n)
-	a := atomic.AddInt32(&w.quota, sz)
-	b := a - sz
-	if b <= 0 && a > 0 {
-		select {
-		case w.ch <- struct{}{}:
-		default:
+	a := atomic.AddInt32(&w.quota, int32(n))
+	wSize := atomic.LoadInt32(&w.waitingSize)
+	if wSize > 0 && a > 0 {
+		for wSize > 0 {
+			select {
+			case w.ch <- struct{}{}:
```

Comment:
Since this is a unblocking write, wouldn't we still end up not broadcasting to all streams waiting for a read on the channel?

Reply:
Sounds good, I'll fix it